include/power/pm_policy.h

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2020 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_POWER_PM_POLICY_H_
+#define ZEPHYR_INCLUDE_POWER_PM_POLICY_H_
+
+#include <power/power.h>
+
+#include <string.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief PM Policy definition
+ *
+ * Store information about a power management policy.
+ */
+struct pm_policy {
+	int (*next_state_get)(struct pm_policy *policy,
+			      int32_t ticks, struct pm_state_info *info);
+	int (*low_power_devices_get)(struct pm_policy *policy,
+				     enum pm_state state);
+};
+
+/**
+ * @brief Statically define and register a power management policy
+ *
+ * @param policy_name The name of this policy
+ * @param policy_next_state_get Callback to get the next PM state
+ * @param policy_low_power_devices_get Callback to determine whether to put
+ *        or not devices in low power state, given the system PM state
+ */
+#define PM_POLICY_DEFINE(policy_name, policy_next_state_get,		\
+			 policy_low_power_devices_get)			\
+	const struct pm_policy policy_name = {				\
```

Comment:
This should probably be `static` unless there's a reason to make it globally visible.